.changeset/friendly-bananas-smoke.md

```diff
@@ -0,0 +1,19 @@
+---
+"@itwin/presentation-hierarchies-react": minor
+---
+
+Added `onHierarchyLimitExceeded` callback to `useTree` and `useUnifiedSelectionTree` for tracking when hierarchy level exceeds the limit.
+
+```typescript
+import { TreeRenderer, useTree } from "@itwin/presentation-hierarchies-react";
+
+function MyTree(props: MyTreeProps) {
+  const state = useTree({
+    ...props,
+    onHierarchyLimitExceeded: () => {
+      console.log(`Hierarchy limit exceeded`);
+    }
```

Comment:
Looking at this example, it seems it would make sense to get the parent node whose hierarchy level exceeded the limit. Maybe even additional config like instance filter and hierarchy level size limit too?

Reply:
I agree it is worth adding the extra information. Changed.

Reply:
Did you consider passing a props object rather than 3 different arguments? That would allow us to add an additional non-nullable parameter in the future without breaking the API.

Reply:
No, did not consider that we might want to add other information but it definitely does not hurt to future-proof the callback. Updated.